src/argilla/training/base.py

```diff
@@ -0,0 +1,170 @@
+            )
+        elif framework == "transformers":
+            if self._rg_dataset_type not in [rg.DatasetForTokenClassification, rg.DatasetForTextClassification]:
+                raise NotImplementedError("Argilla.training does not support Text2Text tasks yet.")
+            self._trainer = ArgillaTransformersTrainer(
+                record_class=self._rg_dataset_type._RECORD_TYPE,
+                dataset=self.dataset_full_prepared,
+                multi_label=self._multi_label,
+                seed=self._seed,
+                model=self.model,
+            )
+        else:
+            raise NotImplementedError(f"Framework {framework} is not supported.")
+
+        self._logger.warning(self)
+
+    def __repr__(self) -> str:
+        """
+        `trainer.__repr__()` prints out the trainer's parameters and a summary of how to use the trainer
+
+        Returns:
+          The trainer object.
+        """
+        return inspect.cleandoc(
+            f"""
+            ArgillaBaseTrainer info:
+            _________________________________________________________________
+            These baseline params are fixed:
+                dataset: {self._name}
+                task: {self._rg_dataset_type.__name__}
+                multi_label: {self._multi_label}
+                required_fields: {self._required_fields}
+                train_size: {self._train_size}
+                seed: {self._seed}
+
+            {self._trainer.__class__} info:
+            _________________________________________________________________
+            The parameters are configurable via `trainer.update_config()`:
+                {self._trainer}
+
+            Using the trainer:
+            _________________________________________________________________
+            `trainer.train(path)` to train to start training. `path` is the path to save the model automatically.
+            `trainer.predict(text, as_argilla_records=True)` to make predictions.
+            `trainer.save(path)` to save the model manually."""
+        )
+
+    def update_config(self, *args, **kwargs):
+        """
+        It updates the configuration of the trainer, but the parameters depend on the trainer.subclass.
+        """
+        self._trainer.update_config(*args, **kwargs)
+
+    def predict(self, text: Union[List[str], str], as_argilla_records: bool = True):
+        """
+        `predict` takes a string or list of strings and returns a list of dictionaries, each dictionary
+        containing the text, the predicted label, and the confidence score.
+
+        Args:
+          text (Union[List[str], str]): The text to be classified.
+          as_argilla_records (bool): If True, the output will be a list of Argilla records instead of dictionaries. Defaults to True.
+
+        Returns:
+          A list of predictions or Argilla records.
+        """
+        return self._trainer.predict(text, as_argilla_records)
+
+    def train(self, path: str = None):
```

Comment:
Thoughts on `output_dir` instead of `path`? `path` gives me the impression of an "input path" moreso than an output one.